vector/src/main/java/im/vector/fragments/roomwidgets/RoomWidgetFragment.kt

```diff
@@ -0,0 +1,324 @@
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.*
+import android.widget.ProgressBar
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import androidx.lifecycle.Observer
+import butterknife.BindView
+import com.airbnb.mvrx.*
+import im.vector.R
+import im.vector.fragments.VectorBaseMvRxFragment
+import im.vector.ui.themes.ThemeUtils
+import im.vector.util.openUrlInExternalBrowser
+
+class RoomWidgetFragment : VectorBaseMvRxFragment() {
+
+    var mWidgetWebView: WebView? = null
+
+    @BindView(R.id.webview_error_layout)
+    lateinit var errorContainer: ViewGroup
+
+    @BindView(R.id.webview_error_text)
+    lateinit var errorText: TextView
+
+    @BindView(R.id.widget_progress_bar)
+    lateinit var webProgressBar: ProgressBar
+
+    val viewModel: RoomWidgetViewModel by activityViewModel()
+
+    override fun getLayoutResId(): Int = R.layout.fragment_room_widget
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        configureWebView()
+        setHasOptionsMenu(true)
+        viewModel.loadWebURLEvent.observe(this, Observer {
+            it?.getContentIfNotHandled()?.let {
+                mWidgetWebView?.clearHistory()
+                mWidgetWebView?.loadUrl(it)
+                mWidgetWebView?.isVisible = true
+            }
+        })
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        mWidgetWebView = view.findViewById(R.id.widget_web_view)
+    }
+
+    override fun onDestroy() {
+        //Looks like WebView creates memory leaks (?) according to legacy code
+        //We have to take additional steps to clear memory
+        destroyWebView()
+        super.onDestroy()
+    }
+
+    private fun destroyWebView() {
+
+        val webview = mWidgetWebView ?: return
+        // Make sure you remove the WebView from its parent view before doing anything.
+        (webview.parent as? ViewGroup)?.removeAllViews()
+
+        webview.webViewClient = null
+        webview.clearHistory()
+
+        // NOTE: clears RAM cache, if you pass true, it will also clear the disk cache.
+        webview.clearCache(true)
+
+        // Loading a blank page is optional, but will ensure that the WebView isn't doing anything when you destroy it.
+        webview.loadUrl("about:blank")
+
+        webview.onPause()
+        webview.removeAllViews()
+
+        // NOTE: This pauses JavaScript execution for ALL WebViews,
+        // do not use if you have other WebViews still alive.
+        // If you create another WebView after calling this,
+        // make sure to call mWebView.resumeTimers().
+        webview.pauseTimers()
```

Comment:
Is it not causing pb with other WebView in the app? Ex with RTC call, Jitsi, VectorWebViewActivity, Captcha, etc.

Reply:
Former code was already pausing/resuming in the WidgetActivity life cycle methods

Reply:
ok...